build.rs

```diff
@@ -0,0 +1,17 @@
+fn main() {
+    let mut feature_count = 0;
+    if cfg!(feature = "shtc1") {
+        feature_count += 1;
+    }
+    if cfg!(feature = "shtc3") {
+        feature_count += 1;
+    }
+    if cfg!(feature = "generic") {
+        feature_count += 1;
+    }
+    if feature_count != 1 {
+        panic!("\n\nMust select exactly of the supported sensors as target feature in Cargo.toml!\nChoices: 'shtc1', 'shtc3' or 'generic'.\n\nExample:\n\n  shtcx = { version = \"0.1\", features = [\"shtc3\"] }\n\n");
```

Comment:
This is not how features should be used, right? The should be freely combinable additions. See also https://doc.rust-lang.org/cargo/reference/manifest.html#usage-in-packages.

Reply:
The STM32 HALs also use this approach to target specific chips. I thought it would be worth a try to see how well it works out.

Another approach would be the using a generic type argument for the different variants, like in the [rn2xx3 crate](https://github.com/dbrgn/rn2xx3-rs/blob/master/src/lib.rs#L146-L156). Would you prefer that?

Reply:
I'll try it with type arguments. The features are getting a bit hacky in CI.

Reply:
> Another approach would be the using a generic type argument for the different variants, like in the rn2xx3 crate. Would you prefer that?

I guess so, yes. But I'm on my phone currently, so I can't really take a serious look at it :)